pkg/util/retry/retry.go

```diff
@@ -150,7 +151,12 @@ func (r *Retry) NextCh() <-chan time.Time {
 }
 
 // WithMaxAttempts is a helper that runs fn N times and collects the last err.
+// It guarantees fn will run at least once. Otherwise, an error will be returned.
 func WithMaxAttempts(ctx context.Context, opts Options, n int, fn func() error) error {
+	if n <= 0 {
+		return errors.Errorf("max attempts should not be 0 or below")
```

Comment:
In case this ever triggers, it'd likely be helpful to include the value of `n` as part of the message